### DISCUSSION TITLE
How to inspect data cache?

### DISCUSSION THREAD

**Discussion by MangoMarcus** [vercel/next.js#66793] [App Router]
I'm finding Next's caching mechanisms very opaque, which makes them hard to debug and troubleshoot.

I've recently had an issue where using `revalidateTag` doesn't seem to refresh a pre-rendered page on Vercel, I'm guessing there's another layer of caching which should be invalidated but isn't. I haven't been able to resolve it without just clearing the whole data cache on Vercel.

It would be very useful to have some visibility / transparency on what exactly's being cached and when.

There's some means to do this locally by trawling the `.next` directory, but there doesn't seem to be any way on Vercel unless I'm missing something??

**Comment by FrenchMajesty** [comment]
I agree, this would be useful for debugging purposes to know what's being stored in there.

**Comment by DhenPadilla** [comment]
Bump on this? Any known way to inspect build-cache size on Vercel deployments?

**Comment by bonesoul** [comment]
also need this feature to get some bugs fixed.

**Comment by ShaunMWallace** [comment]
This would be very useful

**Comment by asimaranov** [comment]
Up, really need to debug why build cache size is so big

**Comment by sengirab** [comment]
Upping, would be a really nice addition to the Observability in Vercel.

**Comment by stebrg** [comment]
I would need that too, upping
